### ISSUE TITLE
Binary sensor: Motion detected by camera

### ISSUE THREAD

**Issue by antonio-fiol** [esphome/feature-requests#651]
**Describe the problem you have/What new integration you would like**

I would like to be able to act (in HA) upon anything moving in front of the camera.

**Please describe your use case for this integration and alternatives you've tried:**

I have checked the ESPHome and HA documentation and did not find anything that helps: the camera implementation in HA is really limited, and besides I don't think this belongs outside the ESP32.
I have seen some people exploring the concept: https://eloquentarduino.github.io/2020/01/motion-detection-with-esp32-cam-only-arduino-version/
I have seen someone comment on the binary sensor being there: https://www.reddit.com/r/homeassistant/comments/bf1zrw/my_experience_with_esphome_on_home_assistant/elfrfrk/
...but I can't find any way to enable such a binary sensor neither in ESPHome or HA.

**Additional context**

I am aware that other (advanced) processing features such as face detection or QR reading have been discarded in #341 , but firstly motion detection was not covered in that request, and secondly motion detection seems (naively) a less complex problem to deal with, and one that would justify (to avoid using bandwidth when not needed) the local processing.

Technical challenge: Would it be possible to do motion detection on the encoded MJPEG stream without fully decoding it?

**Comment by sergeolkhovik** [comment]
I checked the first link - looks really interesting. I tried to build custom solution with ESP32 and PIR sensor, and it was really unstable. PIR generated many false events, so I decided to postpone that idea.

**Comment by antonio-fiol** [comment]
Oh, I was definitely not talking about using a PIR, but just using the
image data. PIR requires extra hardware and it seems that with the image
alone it would be doable.

**Comment by sergeolkhovik** [comment]
Exactly, with PIR I didn't get right result, and would like to have exactly single camera implementation.

**Comment by jeroenterheerdt** [comment]
has anyone ever looked into this? I found this online https://smartsolutions4home.com/ss4h-sd-smart-doorbell/ and it does everything I need to get rid of my Ring, except for the motion detection...

**Comment by sokolsok** [comment]
You can use the camera as a motion sensor. I haven't checked but it should work

**Comment by fkoteam** [comment]
it would be great. 
a new algoritm: 
https://eloquentarduino.github.io/2020/05/easier-faster-pure-video-esp32-cam-motion-detection/
